### PR TITLE
Any root catgory is saved with parents

### PR DESCRIPTION
**Pull request by rogergros**
- Any root category is saved with a parent category

### CODE REVIEW

**Comment by rogergros** [comment]
@mmoreram ready to be reviewed :)

File: src/Elcodi/Bundle/ProductBundle/Tests/Functional/EventListener/RootCategoryEventListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    protected $entityManager;
+
+    /**
+     * Setup
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->categoryRepository = $this
```
Comment (mmoreram) [inline src/Elcodi/Bundle/ProductBundle/Tests/Functional/EventListener/RootCategoryEventListenerTest.php]: In one line. Splitting each `->` in several lines has sense when `count("->") >= 3`

File: src/Elcodi/Component/Product/EventListener/RootCategoryEventListener.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+        return $this;
+    }
+
+    /**
+     * Removes the parent category for a root category.
+     *
+     * @param CategoryInterface $category The category.
+     */
+    protected function removeParentCategoryForRootCategory(
```
Comment (mmoreram) [inline src/Elcodi/Component/Product/EventListener/RootCategoryEventListener.php]: Same here. In one line. This has sense when `count(parameters) >= 2`

**Comment by rogergros** [comment]
Fixed :)

**Comment by mmoreram** [comment]
@rogergros Director :)

**Comment by rogergros** [comment]
Moved to director ^^

**Comment by mmoreram** [comment]
@rogergros Please rebase :)

**Pull request merged**